src/riak_cs_block_server.erl

```diff
@@ -307,24 +309,24 @@ handle_local_notfound(RcPid, FullBucket, FullKey, GetOptions2,
         {error, Why} when Why == disconnected;
                           Why == timeout ->
             _ = lager:debug("get_block_local/5 failed: {error, ~p}", [Why]),
-            RetryFun(NumRetries + 1);
+            RetryFun([{local_quorum, Why}|ErrorReasons]);
 
         {error, notfound} when UseProxyGet andalso ProxyActive->
             case get_block_remote(RcPid, FullBucket, FullKey,
                                   ClusterID, GetOptions2) of
                 {ok, _} = Success ->
                     ProceedFun(Success);
-                {error, _} ->
-                    RetryFun(NumRetries + 1)
+                {error, Reason} ->
+                    RetryFun([{remote_quorum, Reason}|ErrorReasons])
             end;
         {error, notfound} when UseProxyGet ->
-            RetryFun(NumRetries + 1);
+            RetryFun([{local_quorum, notfound}|ErrorReasons]);
         {error, notfound} ->
-            RetryFun(failure);
+            RetryFun([{local_quorum, notfound}|ErrorReasons]);
 
         {error, Other} ->
             _ = lager:error("do_get_block: other error 2: ~p\n", [Other]),
-            RetryFun(failure)
+            RetryFun([{local_quorum, Other}|ErrorReasons])
```

Comment:
Ditto. Except connection error (disconnect) or timeout, retry won't recover the situation, I guess.